src/screens/ShareTrip/ShareTrip.tsx

```diff
@@ -56,48 +186,69 @@ export default function ShareTrip(props: Props): JSX.Element {
                 }
             />
             <ScrollView style={styles.scrollView}>
-                {trip.description && <Text h4>{trip.description}</Text>}
+                {trip.trip.description && <Text h4>{trip.trip.description}</Text>}
                 <SvgLogo style={styles.logo} height={150} width={150} />
-                <Button
-                    containerStyle={styles.btnContainer}
-                    buttonStyle={styles.shareBtn}
-                    title={t("screens.shareTrip.share")}
-                    titleStyle={styles.btnTextStyle}
-                    onPress={() => handleShare(getInvitationLinkPromise(), t)}
-                    loading={loading}
-                />
+                {mode === Mode.SHARE_TRIP && (
+                    <Button
+                        containerStyle={styles.btnContainer}
+                        buttonStyle={styles.shareBtn}
+                        title={t("screens.shareTrip.share")}
+                        titleStyle={styles.btnTextStyle}
+                        onPress={() => handleSubmitButton()}
+                        loading={loadingCreateInvitation}
+                    />
+                )}
                 <Button
                     containerStyle={styles.btnContainer}
                     buttonStyle={styles.planTripBtn}
-                    title={t("screens.shareTrip.planTrip")}
+                    title={t(
+                        mode === Mode.SHARE_TRIP
+                            ? "screens.shareTrip.planTrip"
+                            : "screens.shareTrip.joinTrip"
+                    )}
                     titleStyle={styles.btnTextStyle}
-                    onPress={() =>
-                        props.navigation.reset({
-                            index: 1,
-                            routes: [
-                                {
-                                    name: Routes.DASHBOARD,
-                                },
-                                {
-                                    name: Routes.ITINERARY,
-                                    params: {
-                                        screen: Routes.ITINERARY,
-                                        params: {
-                                            tripId: trip.id,
-                                            tripName: trip.name,
-                                        },
+                    // TODO - Redirect to trip itinerary screen
```

Comment:
I removed this outdated TODO in #75. Probably got re-introduced when merging/rebasing.

```suggestion
```